### DISCUSSION TITLE
[2021.08.23] 해시테이블 구현하기

### DISCUSSION THREAD

**Discussion by pyunji** [pyunji/Hyundai-ITnE-Team_QuadCore#15] [Algorithm]
# 아래에 주어진 조건을 통해 해시테이블을 구현하라
### Node 클래스: key와 value를 가지는 Node 객체를 만드는 클래스

- 생성자
  - param: key - String
  - param: value - String
  - value의 getter/setter
### HashTable 클래스: 
- 생성자
  - param: size - Node를 저장하는 LinkedList 배열의 크기 지정
- getHashCode 메서드: 키를 받아서 문자열 각 자리에 해당하는 아스키코드 값을 모두 더한 값을 해시코드로 반환
	- param: key
	- returns: 해시코드를 정수로 반환
- convertToIndex 메서드: 해시코드를 받아서 인덱스로 반환
	- param: hashcode
- searchKey 메서드: 배열 방의 노드가 여러개 존재하는 경우 key를 이용해 특정 노드를 반환
	- param: list - 인덱스에 해당하는 노드들
	- param: key
	- returns: Node를 반환

- put 메서드: key와 value를 이용해 데이터를 저장하는 함수
	- param: key
	- param: value
- get 메서드: key를 가지고 value를 반환하는 함수
	- param: key

#### ❗ [동영상 참고](https://www.youtube.com/watch?v=Vi0hauJemxA)

**Comment by chaeyeon4u** [comment]
[Hashtable Basic](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/master/ycy/homework/dataStructure/HashTableBasic.java)

**Comment by pyunji** [comment]
[Hashtable Basic](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/pyunji/pyunji/DataStructure/Basic_Hash.java)

**Comment by leechang4** [comment]
[Hashtable Basic](https://github.com/leechang4/Hyundai-ITnE-Team_QuadCore/blob/master/lch/DataStructure/Basic_Hash.java)

**Comment by hyk339** [comment]
[Hashtable Basic Structure](https://github.com/hyk339/Hyundai-ITnE-Team_QuadCore/commit/2f4336f3cb7b7d3821bf42fc96555eeceb72312a)
